src/LinkDotNet.NCronJob/RetryPolicies/RetryHandler.cs

```diff
@@ -48,11 +53,15 @@ public async Task ExecuteAsync(Func<CancellationToken, Task> operation, JobExecu
             var retryPolicy = retryPolicyAttribute?.CreatePolicy(serviceProvider) ?? Policy.NoOpAsync();
 
             // Execute the operation using the given retry policy
-            await retryPolicy.ExecuteAsync(() =>
+            await retryPolicy.ExecuteAsync((ct) =>
             {
                 runContext.Attempts++;
-                return operation(cancellationToken);
-            });
+                return operation(ct);
+            }, cancellationToken).ConfigureAwait(false);
+        }
+        catch (OperationCanceledException)
+        {
+            LogCancellationOperationInJob();
```

Comment:
We don't want to rethrow here?

Reply:
In this context, we handle `OperationCanceledException `without rethrowing it because cancellations are expected behavior, not errors. The distinction is important because:

1. Expected Behavior: Cancellations are a normal part of the lifecycle of tasks that support interruption, not unexpected events that need escalation.
2. Control Flow Management: By catching and logging the cancellation, we manage the workflow gracefully, confirming that the cancellation was detected. We'll add additional processing once states are in place (via events or something).
3. Avoiding Unnecessary Error Handling: Not rethrowing prevents upstream code at the `JobExecutor` from having to handle expected cancellations as errors, simplifying the `JobExecutor` responsibilities.

So the approach avoids conflating cancellations with genuine errors, keeping the system's error handling clean and setting up for when we add states.